refactor(mcp): divide each framework in different files and separate the logic for STDIO and SSE

ELC opened the pull request:
This PR was originally part of #104 but I divide the MCP's specific changes here for clearer review. 

Main points of this PR:
1. Use one file per framework, allowing for Open/Close princpiple when adding new frameworks without modifying existing files.
2. Separate the STDIO and the SSE logic in separate methods and use a facade pattern through the base `setup_tools` method to allow easier expansion of MCP implementations in the future. e.g. if in the future there are websocket MCPs, the addition would be easier to reason about.
3. Replace the try/except for the imports with a `suppress` as it better reflects the intention semantically

ELC commented:
@njbrake Could you help me troubleshoot the failing tests? They seem to be all SSE related, are we missing a mock somewhere?

ELC commented:
@njbrake It was a false alarm, I needed to update the patches in the unit tests

njbrake reviewed (changes requested):
Thanks for the contribution! I like the split up of the MCP files into separate files, makes it clean when updating. I think the logic about handling missing imports needs to be fixed since I think it's a change to the logic. See my comment in the PR for more details

njbrake commented on src/any_agent/__init__.py:
```diff
@@ -1,4 +1,4 @@
 from .frameworks.any_agent import AnyAgent
 
-__all__ = ["AnyAgent", "AgentConfig", "AgentFramework"]
+__all__ = ["AgentConfig", "AgentFramework", "AnyAgent"]
```
Nvm: I see now. It's because we're not running `ruff check . --fix`, which catches this error

ELC commented on src/any_agent/tools/mcp/agno.py:
```diff
@@ -0,0 +1,55 @@
+
+from .mcp_server_base import MCPServerBase
+
+with suppress(ImportError):
+    from agno.tools.mcp import MCPTools as AgnoMCPTools
+    from mcp import ClientSession
+    from mcp.client.sse import sse_client
```
Good catch, I can add that to all the files

ELC commented on src/any_agent/tools/mcp/mcp_server_base.py:
```diff
@@ -0,0 +1,47 @@
+
+mcp_available = False
+with suppress(ImportError):
+    mcp_available = True
```
Done

ELC commented:
@njbrake Changes were made, but now I'm surfacing the precise library in each case based on the framework, a little different than what's in main but I believe addresses the comment

njbrake commented:
> @njbrake Changes were made, but now I'm surfacing the precise library in each case based on the framework, a little different than what's in main but I believe addresses the comment

Thanks! There's a slight tweak still needed to pass the right "lib needed" string back up to the base class, I'll make a few tweaks but I think it's looking good 🙌

njbrake reviewed (approved):
Big fan of the re-org. Thank you 🙏
